### DISCUSSION TITLE
Direct Solvers in Ginkgo

### DISCUSSION THREAD

**Discussion by rtali-ia** [ginkgo-project/ginkgo#1608] [Q&A]
I see that you have an experimental direct solver too /solver/direct.cpp. Is there a sample code that you could share with me that uses your direct solver? Thank you

**Comment by pratikvn** [comment]
We dont have an explicit example yet, but the setup is similar to the [`simple_solver` example](https://github.com/ginkgo-project/ginkgo/blob/develop/examples/simple-solver/simple-solver.cpp). You would need to create a direct solver factory that you want to use, instead of the iterative solver.

For example, to use the direct solver with a LU factorization,

```c++
    using solver_type =
        gko::experimental::solver::Direct<value_type, index_type>;
    using factorization_type =
        gko::experimental::factorization::Lu<value_type, index_type>;

    auto solver_gen =
        solver_type::build()
            .with_factorization(
                factorization_type::build().with_symbolic_algorithm(
                    gko::experimental::factorization::symbolic_type::general))
            .with_num_rhs(static_cast<gko::size_type>(nrhs))
            .on(exec);
```

The rest of the process is same as that in the simple-solver example. You generate it on the matrix you need, `solver_gen->generate(mat)` and then apply to the rhs.
